src/libraries/System.Transactions/src/System/Transactions/Oletx/OletxCommittableTransaction.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Configuration;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+// using System.Runtime.Remoting.Messaging;
+using System.Runtime.Serialization;
+using System.Security.Permissions;
+using System.Threading;
+using System.Transactions.Diagnostics;
+
+#nullable disable
+
+namespace System.Transactions.Oletx
+{
+    /// <summary>
+    /// A Transaction object represents a single transaction.  It is created by TransactionManager
+    /// objects through CreateTransaction or UnmarshalTransaction.  Alternatively, the static Create
+    /// methodis provided, which creates a "default" TransactionManager and requests that it create
+    /// a new transaction with default values.  A transaction can only be committed by
+    /// the client application that created the transaction.  If a client application wishes to allow
+    /// access to the transaction by multiple threads, but wants to prevent those other threads from
+    /// committing the transaction, the application can make a "clone" of the transaction.  Transaction
+    /// clones have the same capabilities as the original transaction, except for the ability to commit
+    /// the transaction.
+    /// </summary>
+    [Serializable]
```

Comment:
I expect we'll want to strip out most or all of the serialization-related support.
cc: @GrabYourPitchforks

Reply:
Yep. I still need to do some research, but from what I understand, System.Transactions had functionality where attempting to send a Transaction across AppDomain boundaries would automatically trigger escalation to a distributed transaction, and then propagate the transaction via OleTx. I think this relied on BinaryFormatter, and I definitely don't think we need to do this.

We can always see if we get user demand in the future for this (I doubt it), and if so, reimplement this in some safer way.